src/components/Search/SearchPageHeader.tsx

```diff
@@ -163,7 +164,8 @@ function SearchPageHeader({queryJSON, hash, onSelectDeleteOption, setOfflineModa
             },
         });
 
-        const shouldShowHoldOption = !isOffline && selectedTransactionsKeys.every((id) => selectedTransactions[id].canHold);
+        const shouldShowHoldOption =
+            !isOffline && selectedTransactionsKeys.every((id) => selectedTransactions[id].canHold) && TransactionUtils.areRequestsInTransactionsUnsubmitted(selectedTransactionsKeys);
```

Comment:
Selected transactions should be all submitted to show the `hold` option right?
```suggestion
            !isOffline && selectedTransactionsKeys.every((id) => selectedTransactions[id].canHold) && !TransactionUtils.areRequestsInTransactionsUnsubmitted(selectedTransactionsKeys);
```

Reply:
Yes, if one of them is unsubmitted then the whole flow should be blocked.

Reply:
@cdOut So, here it should be `!TransactionUtils.areRequestsInTransactionsUnsubmitted(selectedTransactionsKeys)` right?

Reply:
@eh2077 no, because the utili is whether any requests from the transactions are **unsubmitted**. So if it finds one unsubmitted it returns `true`. The logic is correct in the current implementation.

Reply:
@eh2077 That's correct, I've added the change and retested, it works properly. Must've missed it while moving it between components.